### ISSUE TITLE
x17 occasional crash when mining

### ISSUE THREAD

**Issue by joeyoon80** [Sniffdog/Sniffdogminer#33]
Hi. I am facing an occasional crash when mining x17 using your software in ahashpool. I checked the event viewer, and here is the error detail:

Faulting application name: ccminer.exe, version: 1.0.0.0, time stamp: 0x58d6ee8f
Faulting module name: ccminer.exe, version: 1.0.0.0, time stamp: 0x58d6ee8f
Exception code: 0xc0000005
Fault offset: 0x000000000000ea81
Faulting process id: 0xb3c
Faulting application start time: 0x01d37ad7e52701c8
Faulting application path: C:\Users\Joe Black\Desktop\Sniffdogminer 3.0\Bin\NVIDIA-Alexis\ccminer.exe
Faulting module path: C:\Users\Joe Black\Desktop\Sniffdogminer 3.0\Bin\NVIDIA-Alexis\ccminer.exe
Report Id: 0719a0ea-4e8f-44ab-85e0-828d618d5bce
Faulting package full name: 
Faulting package-relative application ID: 

My mining rig is using the following:

Intel Pentium G4600
Biostar TB250-BTC Pro
3x Asus Strix 1070 o8g
8GB RAM
120GB SSD

Some software details:
Nvidia driver version: 388.43
Windows 10 Pro 64-bit
MSI Afterburner 4.4.0
Virtual Memory: 64GB

Please assist.

**Comment by Sniffdog** [comment]
yeah we had many people come to us with overclocked machines and I try to tell them this program is not setup for overclocking

**Comment by joeyoon80** [comment]
Ah okay, yes my rig is overclocked, and I find that only a few algos that were crashing (x17,sib,hsr).

For the setting u mentioned to modify, may I know what is it used for and what should I refer to when modifying it?

Alternatively, if i completely turn off the overclocking software, should it solve the problem? Just to be aware that my gfx cards come already factory overclocked.

**Comment by Sniffdog** [comment]
yes if you take off the -i settings the miner will use default settings... Just be carefull overclocking... All these algo's run on different capacities. For example your Virtual memory should be min 16000 for sib max 64000

**Comment by joeyoon80** [comment]
Noted, will do that. Thanks. For my overclocking, i have checked numerous references and have only overclocked to 50% of its full tested overclocking potential. Hopefully its not an issue.

**Comment by Sniffdog** [comment]
Yeah we don't run with any overclocking 8 card 1080ti rigs. We placed the intensity setting onto Ccmineralexis for more than 3 card rigs

**Comment by joeyoon80** [comment]
Noted. I had not thought of modifying the files that u guys published. Will look into it and adjust accordingly. Thanks again for the assistance and pudate. :)

**Comment by joeyoon80** [comment]
[2017-12-25 11:26:05] GPU#0:ASUS GTX 1070, 86.76MH/s
[2017-12-25 11:26:05] GPU#0:ASUS GTX 1070, 0.58MH/W, 0.044MH/Mhz
[2017-12-25 11:26:05] GPU#0:ASUS GTX 1070, 53C(F:53%) 1967/4404MHz(150W)
[2017-12-25 11:26:05] GPU#0:an illegal memory access was encountered
Cuda error in func 'quark_blake512_cpu_setBlock_80' at line 547 : an illegal memory access was encountered.

Hi. After tweaking as per your suggestion. I still face occasional issue where x17 mining with ccminer crashing. The above was the last line at the console.

**Comment by Sniffdog** [comment]
yeah I left for 8 days... had all my rigs to re-boot every 24 hours... and placed Sniffdog as a startup program...

I never once had a problem so I'm thinking (and assuming) you have a hardware issue...

**Comment by joeyoon80** [comment]
Okay, I have done some tinkering, updated all the drivers. I believe its my bad, as I think the issue boils down to the custom overclocking I set for my GPUs. After turning off all the overclocking (and just leaving the custom fanspeed adjustment), I had been running past 24 hours without any errors or crashes. I will continue to monitor the issue and see if it has been resolved, and will close the ticket in the next 48 hours.

Just to ask further, does Delay function apply in this software like Multipoolminer?

**Comment by Sniffdog** [comment]
yes it does and allows the cards to cool better between algos

**Comment by joeyoon80** [comment]
Ok, 48 hours passed, no issues whatsoever. Just before I close this issue, just a final question regarding setting it to reboot on intervals. The windows account I am using has a password because I use RDP as my remote access. I have tried the following:

1) Setup scheduled task to reboot every 24 hours, then setup to run startsniffin.bat upon startup, but doesnt start.
2) Copy startsniffin.bat shortcut into Startup folder, but only runs when account logged in.

Is there a way for the program to startup upon reboot without login to windows?

**Comment by Sniffdog** [comment]
i have had to unlock that feature in windows in order for the startup programs to work...

https://www.tekrevue.com/tip/skip-windows-10-login-screen/

**Comment by joeyoon80** [comment]
ok, thanks for the tip. I will close this issue. thanks again for all the assistance.

**Comment by tbranch227** [comment]
Would adjusting the power target on the GPU's cause this miner to be unstable?

**Comment by joeyoon80** [comment]
unsure, but definitely overclocking does cause it to crash randomly. now my MSI afterburner only tunes the custom fan curve, but all clocks and power targets are left at default, and stable for many days already wit daily restarts as per recommended.

**Comment by kingyem** [comment]
I've done undervolting, with slight core overclock and memory overclock for my system and the miner has been stable

**Comment by jeremyofmany** [comment]
I was using SniffDog 4.2 to mine x17 and other algos without issue (even with a +50 MHz offset on the GPU Clock using Precision XOC (EVGA) and had no issues whatsoever.

I updated to SniffDog 4.3 and ccmineralexis began crashing on x17 only.
EVGA 1080 Ti FTW3 (manufacturer pre-overclocked)
Windows 7 x64
I updated PowerShell to 6
I updated my NVIDIA drivers to latest.
I deleted the entire SniffDog folder (saving my startsniffin.bat file), re-extracted and even benchmarking again, x17 crashes.
I reset my GPU to stock settings (no +50 MHz GPU Clock offset)
I edited the Ccmineralexis.ps1 file and changed this line:
X17 = ' -i 21.5 --api-remote'
to this:
X17 = ' --api-remote'

Even after reverting back to SniffDog 4.2, the problem persists, so I am wondering if that's because I have updated PowerShell and my NVIDIA drivers to coincide with 4.3.

When the miner starts, it stops and then gets stuck in an infinite loop of trying to restart.

Thanks.
